packages/api/tests/lambdas/test-replay-sqs-messages.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const test = require('ava');
+const uuidv4 = require('uuid/v4');
+
+const { deleteQueue } = require('@cumulus/aws-client/SQS');
+const { createBucket, recursivelyDeleteS3Bucket, s3PutObject } = require('@cumulus/aws-client/S3');
+const { getS3KeyForArchivedMessage } = require('@cumulus/ingest/sqs');
+const { randomString } = require('@cumulus/common/test-utils');
+
+const { createSqsQueues, getSqsQueueMessageCounts } = require('../../lib/testUtils');
+const {
+  getArchivedMessagesFromQueue,
+  replaySqsMessages,
+} = require('../../lambdas/replay-sqs-messages');
+
+test.before(async (t) => {
+  process.env.system_bucket = randomString();
+  t.context.system_bucket = process.env.system_bucket;
+  await createBucket(t.context.system_bucket);
+});
+
+test.beforeEach(async (t) => {
+  process.env.stackName = 'test-stack';
+  t.context.stackName = process.env.stackName;
+
+  const queues = await createSqsQueues(randomString());
+  t.context.queueUrl = queues.queueUrl;
+  t.context.queueName = queues.queueName;
+
+  const validMessage = { id: uuidv4(), Body: JSON.stringify({ testdata: randomString() }) };
+  t.context.validMessage = validMessage;
+  const key1 = getS3KeyForArchivedMessage(t.context.stackName, validMessage.id, queues.queueName);
+  await s3PutObject({
+    Bucket: process.env.system_bucket,
+    Key: key1,
+    Body: validMessage.Body,
+  });
+});
+
+test.afterEach(async (t) => {
+  await deleteQueue(t.context.queueUrl);
+});
+
+test.after(async (t) => {
+  await recursivelyDeleteS3Bucket(t.context.system_bucket);
+});
+
+test('replaySqsMessages queues messages to SQS for each archived message', async (t) => {
+  const { queueName, queueUrl, validMessage } = t.context;
+
+  const event = {
+    queueName,
+  };
+  const expected = [JSON.parse(validMessage.Body)];
+
+  const replay = (await replaySqsMessages(event));
+  const {
+    numberOfMessagesAvailable,
+    numberOfMessagesNotVisible,
+  } = await getSqsQueueMessageCounts(queueUrl);
+  t.is(numberOfMessagesAvailable, 1);
+  t.is(numberOfMessagesNotVisible, 0);
+  t.deepEqual(replay, expected);
+});
+
+test('replaySqsMessages queues only valid messages to SQS', async (t) => {
```

Comment:
I think this is a better test of `getArchivedMessagesFromQueue`, because I think you're proving that a non-JSON message won't get queued. But the logic that does the JSON parsing of the S3 object is in `getArchivedMessagesFromQueue`. So if this test were failing, that's where the fix should be.

Reply:
Good point [`030ed97` (#2373)](https://github.com/nasa/cumulus/pull/2373/commits/030ed97f35a8892454a2f46e229a87b2bb447203#diff-6d4399c4bcca5a24555c1881c0ceb23928bb79136abba53f19f2fc254b0d976eL67)

Reply:
So you updated the test name but you didn't really change the test logic. We should update the test so that it calls and is testing `getArchivedMessagesFromQueue`  directly, not via `replaySqsMessages`

Reply:
I think I misunderstood. Should be updated now [`442ed82` (#2373)](https://github.com/nasa/cumulus/pull/2373/commits/442ed821f55ec235fa6fac74264716f11c966844)